Tests/Recurly/Addon_Test.php

```diff
@@ -59,4 +59,26 @@ public function testCreateXmlTieredAddOn() {
 
     $this->assertEquals("<?xml version=\"1.0\"?>\n<add_on><add_on_code>little_llama</add_on_code><tier_type>tiered</tier_type><tiers><tier><unit_amount_in_cents><USD>400</USD></unit_amount_in_cents><ending_quantity>800</ending_quantity></tier><tier><unit_amount_in_cents><USD>200</USD></unit_amount_in_cents></tier></tiers></add_on>\n", $addon->xml());
   }
+
+  public function testCreateXmlPercentageTieredAddOn() {
+    $item = new Recurly_Item();
+    $addon = new Recurly_Addon();
+    $addon->plan_code = 'gold';
+    $addon->add_on_code = 'little_llama';
+    $addon->tier_type = 'tiered';
+
+    $percentage_tier = new Recurly_PercentageTier();
+    $percentage_tier->currency = 'USD';
+    $tier = new Recurly_PercentageTierByCurrency();
+    $tier->ending_amount_in_cents = 100;
+    $tier->usage_percentage = '10.0';
+    $tier2 = new Recurly_PercentageTierByCurrency();
+    $tier2->ending_amount_in_cents = null;
+    $tier2->usage_percentage = '20.0';
+    $percentage_tier->tiers = array($tier, $tier2);
+
+    $addon->percentage_tiers = array($percentage_tier);
+
+    $this->assertEquals("<?xml version=\"1.0\"?>\n<add_on><add_on_code>little_llama</add_on_code><tier_type>tiered</tier_type><percentage_tiers><percentage_tier><currency>USD</currency><tiers><tier><ending_amount_in_cents>100</ending_amount_in_cents><usage_percentage>10.0</usage_percentage></tier><tier><ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents><usage_percentage>20.0</usage_percentage></tier></tiers></percentage_tier></percentage_tiers></add_on>\n", $addon->xml());
+  }
 }
```

Comment:
I’m curious, it it possible (and easy) to format this to be more readable?
```php
$this->assertEquals("<?xml version=\"1.0\"?>\n<add_on><add_on_code>little_llama</add_on_code><tier_type>tiered</tier_type><percentage_tiers><percentage_tier><currency>USD</currency><tiers><tier><ending_amount_in_cents>100</ending_amount_in_cents><usage_percentage>10.0</usage_percentage></tier><tier><ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents><usage_percentage>20.0</usage_percentage></tier></tiers></percentage_tier></percentage_tiers></add_on>\n", $addon->xml());
```

something like:
```php
$this->assertEquals("
  <?xml version=\"1.0\"?>\n<add_on>
    <add_on_code>little_llama</add_on_code>
    <tier_type>tiered</tier_type>
    <percentage_tiers>
      <percentage_tier>
        <currency>USD</currency>
        <tiers>
          <tier>
            <ending_amount_in_cents>100</ending_amount_in_cents>
            <usage_percentage>10.0</usage_percentage>
          </tier>
          <tier>
            <ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents>
            <usage_percentage>20.0</usage_percentage>
          </tier>
        </tiers>
      </percentage_tier>
    </percentage_tiers>
  </add_on>
", $addon->xml());
```

Reply:
Yes is possible if we change assertEquals to assertXmlStringEqualsXmlString. I changed that.